b1ddi/resource_ipam_subnet.go

```diff
@@ -595,14 +602,16 @@ func resourceIpamsvcSubnetRead(ctx context.Context, d *schema.ResourceData, m in
 func resourceIpamsvcSubnetUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	c := m.(*b1ddiclient.Client)
 
-	if d.HasChange("address") {
-		d.Partial(true)
-		return diag.FromErr(fmt.Errorf("changing the value of 'address' field is not allowed"))
-	}
+	if !d.IsNewResource() {
+		if d.HasChange("address") || d.HasChange("parent") {
+			d.Partial(true)
+			return diag.FromErr(fmt.Errorf("changing the value of 'address' or 'parent' field is not allowed"))
```

Comment:
Shouldn't you use https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/diag#Errorf function to build new errors?

Reply:
I left the old cold as is, it used fmt package. Will update to use diag.Errorf